docs/design/adminconsole/WORKER-CRON-PROCEES.md

```diff
@@ -0,0 +1,202 @@
+
+#### Schedule within the executable
+
+Scheduling within the executable means managing the schedule from within the application itself. While this makes setup locally simple, this approach that has some immediate obvious downsides. The application owner must manage the scheduler and must do so within the execution context of the application. Also, the container would have to run continuously, so the user would need to exercise care managing resources for a long running application process.
+
+#### Schedule controlled by the host system / orchestrator
+
+This approach is more common in container orchestration environments where a control plane can manage execution schedules. The ephemeral nature of containers make them a great environment for tasks that need to short execution times. The downside is that the configuration for this requires a bit more overhead and configuration beyond the host system and executable, which might not be ideal in scenarios where simplicity and speed of deployment are highly valued.
+
+#### Schedule controlled within container
+
+This approach is a hybrid of the two previous approaches, taking advantages and benefits from both. The advantage here is that the Docker image running in a container environment, can be built using a multi-stage docker file, which allows the entry point of the image to be defined at build time. This allows us to take advantage of the `crontab` process, which provides a configuration for 1) defining a process and 2) scheduling execution of that process. In this case, this process is the published console application, and can be configured to run on a schedule using a `crontab` file.
+
+### Recommendation
+
+After careful consideration, the approach of managing Cron within the Docker Image itself appears to be the best fit for our use case. This will provide customizable scheduling via a configurable file while allowing the execution to remain in the container environment, increasing security and testability.
+
+The steps for implementing the recommendation are provided below:
+
+#### Prepare the App Context
+
+The Docker file uses `WORKDIR /source` to reference the root of the application. That is to say `/source/Application/EdFi.AdminConsole.HealthCheck.Service` points to `/[your_local_repo_dir]/Application/EdFi.AdminConsole.HealthCheck.Service`
+
+The Docker file will also need to support exposing the configuration for the following environment variables:
+
+```
+{
+    "AdminApiSettings": {
+      "ApiUrl": "https://localhost:7218/",
+      "AdminConsoleInstancesURI": "adminconsole/instances",
+      "AdminConsoleHealthCheckURI": "adminconsole/healthcheck",
+      "AccessTokenUrl": "https://localhost:7218/connect/token"
+  }
+}
+```
+
+These can be exposed either by using the `AppSettingVariable__NestedKey=VALUE` accessor pattern to expose the values to the container.
+
+The `ClientId` and `ClientSecret` values will need to be retrieved from upon instance creation.
```

Comment:
Note: there are two different sets of credentials:

* For either worker to access Admin API
* For Health Check to access ODS/API

Thes second set can be retrieved at runtime from Admin API. App settings or env var will need to provide the Admin API credentials to go along with the `AccessTokenUrl`.